jax/ops/scatter.py

```diff
@@ -308,3 +309,35 @@ def segment_sum(data, segment_ids, num_segments=None):
   out = jnp.zeros((num_segments,) + data.shape[1:], dtype=data.dtype)
   segment_ids = jnp.mod(segment_ids, num_segments)
   return index_add(out, segment_ids, data)
+
+
+def segment_max(data, segment_ids, num_segments=None):
+  """Computes the max within segments of an array.
+
+  Similar to TensorFlow's segment_max:
+  https://www.tensorflow.org/api_docs/python/tf/math/segment_max
```

Comment:
You may need a blank line before `Args`.

I wanted to check the documentation rendering and then I realized this PR also needs to add `segment_max` to the docs. (`jax.ops.rst`)